lib/Cro/HTTP/Client.pm6

```diff
@@ -515,7 +517,7 @@ class Cro::HTTP::Client {
 
         my $parsed-url = self && $!base-uri
             ?? $!base-uri.add($url)
-            !! Cro::Uri::HTTP.parse($url);
+            !! Cro::Iri::HTTP.parse($url).to-uri;
```

Comment:
I wonder if we're reparsing here too, and if we should perhaps be calling the `wrap-uri` logic (lifted a scope up) instead?